internal/component/pyroscope/write/write.go

```diff
@@ -466,3 +476,36 @@ func ensureNameMatchesService(lbls labels.Labels) labels.Labels {
 	}
 	return lbls
 }
+
+// validateLabels checks for valid labels and doesn't contain duplicates.
+func validateLabels(lbls labels.Labels) error {
+	if lbls.Len() == 0 {
+		return labelset.ErrServiceNameIsRequired
+	}
+
+	seen := make(map[string]struct{}, lbls.Len())
```

Comment:
Nit: You could also go for the approach of sorting the label slice by name and then checking for duplicates like the backend.

Reply:
Done in 4879713eec22cce3535dc2ea7f59a778f98f5836